fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/LAORepository.java

```diff
@@ -116,4 +117,29 @@ public void updateNodes(Channel channel) {
   public Map<MessageID, MessageGeneral> getMessageById() {
     return messageById;
   }
+
+  public Optional<LaoView> getLao(String id) {
+    if (laoById.containsKey(id)) {
+      return Optional.of(new LaoView(laoById.get(id).getLao()));
+    }
+    return Optional.empty();
+  }
+
+  public Optional<LaoView> getLaoViewByChannel(Channel channel) {
+    return getLao(channel.extractLaoId());
+  }
+
+  public void updateLao(Lao lao) {
+    if (lao == null) {
+      throw new IllegalArgumentException();
+    }
+
+    if (laoById.containsKey(lao.getId())) {
+      // If the lao already exists, we can push the next update
+      laoById.get(lao.getId()).publish(lao);
+    } else {
+      // Otherwise, create the state
+      laoById.put(lao.getId(), new LAOState(lao));
+    }
```

Comment:
is this call supposed to be thread-safe ? If so, this `laoById` access wouldn't work without some form of locking.